tutorials/fundamentals/filter-key.md

```diff
@@ -0,0 +1,36 @@
+To filter an object using the keys, you can use JavaScript's `Object.hasOwnProperty()` method.
+
+```javascript
+const obj = { name: 'Masteringjs', location: 'Florida', help: true };
+let newObj = {}
+if(obj.hasOwnProperty('name')) {
+    newObj['name'] = obj['name'];
+}
+```
+
+## Object.keys()
+
+Another approach to filter using keys is to use the `Object.keys()` method.
+`Object.keys()` will return an array of the keys and you can then iterate through
+and filter accordingly.
+
+```javascript
+const obj = { name: 'Masteringjs', location: 'Florida', help: true };
+let newObj = {}
+let arr = Object.keys(obj);
+for(let i = 0; i < arr.length; i++) {
+    if (arr[i] == 'name') {
```

Comment:
Not a very good example: if you just want to pick out 'name' there's an easier way to do that. A better example would be to return an object containing only the keys that contain 'name' or something similar.

Reply:
Isn't that what this is doing?

Reply:
No, this is returning an object containing only the keys that are _exactly equal to_ name, rather than _contain_ name.